source/tunneling/SecureTunnelingFeature.cpp

```diff
@@ -102,7 +102,7 @@ namespace Aws
                         std::unique_ptr<SecureTunnelingContext> context =
                             unique_ptr<SecureTunnelingContext>(new SecureTunnelingContext(
                                 mSharedCrtResourceManager,
-                                *config.rootCa,
+                                mRootCa,
                                 *config.tunneling.destinationAccessToken,
                                 GetEndpoint(*config.tunneling.region),
                                 static_cast<uint16_t>(config.tunneling.port.value()),
```

Comment:
nit: this isn't too relevant to your PR, but I wonder why sometimes we use the * operator to get the optional value and other times we call value(). Looking into the Aws::Crt::Optional type, it seems like the behavior is the same for both methods of access.

Reply:
Technically equivalent, but sometimes operator precedence rules require wrapping the dereferenced value in parenthesis eg `(*optional_thing)->` This would be most common when the optional thing you have is actually a pointer to something else and you want to use the member access operator `->` on the dereferenced thing.  You can see from the precedence table below that member access is higher precedence than dereference hence you need the parens. In contrast, use of `value()` doesn't require the extra parenthesis.

https://en.cppreference.com/w/cpp/language/operator_precedence